src/nl/Stratego/Bord.java

```diff
@@ -78,6 +78,24 @@ public boolean pieceCheck (int pionYLocation, int pionXLocation, Speler spelerAa
         }
     }
 
+    boolean checkValidPiece(int pionYlocation, int pionXLocation, int team){
+        Object gekozenStuk = speelBord[pionYlocation][pionXLocation];
+        if(gekozenStuk == blokkade){
+            System.out.println("Je hebt een blokkade gekozen");
+            return false;
+        } else if (gekozenStuk == null){
+            System.out.println("Je hebt een lege plek gekozen");
+            return false;
+        } else {
+            Speelstuk gekozenSpeelStuk = (Speelstuk)gekozenStuk; //casten naar een Speelstuk object
+            if (gekozenSpeelStuk.getTeam() != team){
+                System.out.println("het gekozen speelstuk is niet van jou team");
```

Comment:
"het gekozen speelstuk is niet van jouw team"  
je bent de 'w' vergeten achter jou hihi